From 8db783f7f0bdb7b4c9dd7a7bd19f6c7bf0bc4f06 Mon Sep 17 00:00:00 2001
From: Andreas Dangel <andreas.dangel@pmd-code.org>
Date: Thu, 14 Jan 2021 18:59:31 +0100
Subject: [PATCH] test case 2 - single xpath rule changed

The rule AvoidMessageDigestField is disabled
and will produce false negatives. This is a XPath rule.

Only the rules from the same category (bestpractices)
should be executed and should be compared against
the baseline.
---
 pmd-java/src/main/resources/category/java/bestpractices.xml    | 2 +-
 .../java/rule/bestpractices/AvoidMessageDigestFieldTest.java   | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/pmd-java/src/main/resources/category/java/bestpractices.xml b/pmd-java/src/main/resources/category/java/bestpractices.xml
index 59e5e56c72..c29a13748e 100644
--- a/pmd-java/src/main/resources/category/java/bestpractices.xml
+++ b/pmd-java/src/main/resources/category/java/bestpractices.xml
@@ -140,7 +140,7 @@ public class Foo {
             <property name="xpath">
                 <value>
                     <![CDATA[
-//FieldDeclaration[pmd-java:typeIs('java.security.MessageDigest')]
+//FieldDeclaration[pmd-java:typeIs('java.security.MessageDigestFoo')]
 ]]>
                 </value>
             </property>
diff --git a/pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AvoidMessageDigestFieldTest.java b/pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AvoidMessageDigestFieldTest.java
index 37d37625e7..ef3c8aa5c6 100644
--- a/pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AvoidMessageDigestFieldTest.java
+++ b/pmd-java/src/test/java/net/sourceforge/pmd/lang/java/rule/bestpractices/AvoidMessageDigestFieldTest.java
@@ -4,8 +4,11 @@
 
 package net.sourceforge.pmd.lang.java.rule.bestpractices;
 
+import org.junit.Ignore;
+
 import net.sourceforge.pmd.testframework.PmdRuleTst;
 
+@Ignore
 public class AvoidMessageDigestFieldTest extends PmdRuleTst {
     // no additional unit tests
 }
-- 
2.29.2

